### PR TITLE
Hook search up to results

### PR DESCRIPTION
**Pull request by dannycho7**
This PR introduces infrastructure changes to the HOC, Search, and Results component to allow the search component to add filters to the Results table.

What it looks like generally:
image

Results Component receiving the filter term:
image

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -2,19 +2,40 @@ import React, { Component } from "react";
...
     super(props);
+
+    this.state = {
+      term: ""
+    };
+
+    this.onSearchSubmit = this.onSearchSubmit.bind(this);
+  }
+
+  onSearchSubmit(term) {
```
Comment (estensen) [inline client/src/App.js]: Consider using an arrow function instead. Then you don't need to bind the function.

From Google's JS [styleguide](https://google.github.io/styleguide/jsguide.html#features-functions-arrow-functions):
> Never call f.bind(this) or goog.bind(f, this) (and avoid writing const self = this). All of these can be expressed more clearly and less error-prone with an arrow function.

File: client/src/Results/Result.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import React from "react";
+import { TableRow, TableRowColumn } from 'material-ui/Table';
+import RaisedButton from 'material-ui/RaisedButton';
+
+const Result = ({ name, times}) => (
+	<TableRow>
+		<TableRowColumn>{name}</TableRowColumn>
+		<TableRowColumn>{times}</TableRowColumn>
+		<TableRowColumn>
+			<RaisedButton label="Add" primary />
```
Comment (estensen) [inline client/src/Results/Result.js]: Indent to the same level as `TableRowColumn`

File: client/src/Results/index.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import React, { Component } from "react";
-import { Table, TableBody, TableHeader, TableHeaderColumn, TableRow, TableRowColumn } from 'material-ui/Table';
-import RaisedButton from 'material-ui/RaisedButton';
+import { Table, TableBody, TableHeader, TableHeaderColumn, TableRow } from 'material-ui/Table';
+import Result from "./Result";
 
 class Results extends Component {
   constructor(props) {
```
Comment (estensen) [inline client/src/Results/index.js]: Remove useless constructor

File: client/src/Search/index.js [hunk 4]
```diff
@@ -1,24 +1,41 @@
...
+
+    this.state = {
+      term: ""
+    };
+
+    this.onChange = this.onChange.bind(this);
+    this.onRequestSearch = this.onRequestSearch.bind(this);
+  }
+
+  onChange(term) {
```
Comment (estensen) [inline client/src/Search/index.js]: Consider using an arrow function instead

File: client/src/Search/index.js [hunk 5]
```diff
@@ -1,24 +1,41 @@
...
+    this.setState({ term });
+  }
+
+  onRequestSearch() {
```
Comment (estensen) [inline client/src/Search/index.js]: Consider using an arrow function instead

File: client/src/App.js (hunk 1 above)
Comment (estensen) [reply client/src/App.js]: Bind is also [slower](https://stackoverflow.com/questions/17638305/why-is-bind-slower-than-a-closure/17638540#17638540) than a clouse

File: client/src/Results/index.js (hunk 3 above)
Comment (dannycho7) [reply client/src/Results/index.js]: The constructor will be used when we add interactivity with the calendar

File: client/src/App.js (hunk 1 above)
Comment (dannycho7) [reply client/src/App.js]: The performance difference is minuscule, but I changed it in a new commit

**Pull request merged**